### ISSUE TITLE
Selected processes remain selected

### ISSUE THREAD

**Issue by andre-hohmann** [kitodo/kitodo-production#4579]
### Problem
Selected processes remain in some cases selected, although they have been deselected. 
This leads in some cases to the application of exports, Scripts, ... to unwanted processes. 
It is difficult to search for selected processes. The only way to make sure to deselect all processes is to logout and login again or to run a script - as after the script finished, the processes on which it was applied are listed in the message. 

### Solution
Deselected processes should remain deselected. 

### Example
The following Gif shows that the unselected process is again selected, after the complete process list is shown and the process is shown after filtering for it by its ID. 
 
![markierteVorgänge01](https://user-images.githubusercontent.com/14835770/129318862-4e1f737a-e5a0-4f7a-bf4c-2c18e1b37c3f.gif)

**Comment by solth** [comment]
Related to or duplicate of #3727

**Comment by matthias-ronge** [comment]
Yes, this i a duplicate of #3727, but I tried to reproduce it a while ago and failed, so I closed the other issue. We can keep this issue open.

**Comment by matthias-ronge** [comment]
One requirement should be that whenever the filter box is changed, the selection should be cleared.

**Comment by solth** [comment]
Processes are selected for the whole session atm, so unless the user renews that session by logging out and in again, the process selection will remain, I am afraid.

**Comment by andre-hohmann** [comment]
That is a very dangerous behaviour. I expect that it is impossible to communicate this behaviour to all users and that it will lead to many unwanted actions (export, manipulating metadata, change of task status, ...). 

From my point of view, the deselection of processes is absolutely necessary.

**Comment by solth** [comment]
@andre-hohmann yes, I know its a bug. I didn't want to imply that it is working as intended

**Comment by markusweigelt** [comment]
I cannot reproduce the behavior atm. When I adjust the filter, all processes are deselected. Is the a way to reproduce this behavior or does issue merge #4583 solves described problem?

@solth 
> Processes are selected for the whole session atm, so unless the user renews that session by logging out and in again, the process selection will remain the same and won't reset automatically, I am afraid.

When I refresh the browser window or change the filter, the checkbox is automatically deselected. (Tested with the browser cache turned on/off)

Do you still have the problem in current master version?

**If the problem persists, we need more informations to reproduce the problem**
What permissions does the user have?
Is there an error in the browser console?
What browser are you using?

**Comment by andre-hohmann** [comment]
In the SLUB, the problem still persists. 

What permissions does the user have?
- user 1: administrative permissions
- user 2: Metadata, Scanning, ProcessManagement (we adjusted the permissions)
- i think it does not depend on the permissions

Is there an error in the browser console?
- i cannot answer that

What browser are you using?
- Google Chrome - Version 93.0.4577.63 (Offizieller Build) (64-Bit)

**Comment by matthias-ronge** [comment]
Hi André, the question was, with the latest master, after #4583 was merged three days ago. I don’t know, but I guess “at the SLUB” the software has not been updated within the last three days? Or, can you confirm it was updated?

**Comment by solth** [comment]
@markusweigelt the cause of this problem is already known - it's the `SessionScope` of the `ProcessForm` bean. It leads to all variables in that bean to retain their value until explicitely changed or until the user renews the session by logging out of and back into the system. Instead, the bean should be `ViewScoped` so it is rebuild every time the "view," e.g. the page is opened by the user.

Unfortunately just changing the annotation at the beginning of the class from `SessionScoped` to `ViewScoped` does not sufficiently solve this issue because some methods _rely_ on some values being retained for the whole session (see my comment in #4490 for details - they apply to `ProcessForm` in a similar way as they do to `WorkflowForm`)

**Comment by andre-hohmann** [comment]
> Hi André, the question was, with the latest master, after #4583 was merged three days ago. I don’t know, but I guess “at the SLUB” the software has not been updated within the last three days? Or, can you confirm it was updated?

@matthias-ronge : Your assumption is correct - we **do not use** the newest master. Our version is from last week. Thus, my answer above can be ignored.

**Comment by solth** [comment]
I just tested the current master and the issue does indeed occur less than before. One situation where selected processes remain selected is after a process is opened in the metadata edtior and the user returns to the list via canceling or saving in the editor. Many of the other cases where processes would remain selected seem to be resolved already.

**Comment by andre-hohmann** [comment]
@solth : Is it a problem, when the processes remain selected after closing the metadata editor? 
In my opinion, that is not wrong and maybe even an advantage. 

I tested several cases and think also that it works it now. 
Is it OK to close the issue?

**Comment by andre-hohmann** [comment]
I will close the issue to avoid unnecessary examinations. 
If other bugs are discovered, they can be described in a new issue.
